guiEditor/src/workbenchEditor.tsx

```diff
@@ -55,6 +55,9 @@ export class WorkbenchEditor extends React.Component<IGraphEditorProps, IGraphEd
         if (navigator.userAgent.indexOf("Mobile") !== -1) {
             ((this.props.globalState.hostDocument || document).querySelector(".blocker") as HTMLElement).style.visibility = "visible";
         }
+        this.props.globalState.onPopupClosedObservable.addOnce(() => {
+            this.componentWillUnmount();

```

Comment:
this is usually called by react itself, and should not be called by the developer. Is there a reason this is directly needed?
I assume you need to remove the event listener, the question is if you can do that in an outside function and let react do its cleanup on its own. Or maybe you need to make sure this component (which is the main one if I am not mistaken?) is also unmounted when the popup closes.

Reply:
I'm not sure why this one is necessary, it's just copied from NME. I will check if we can just call the cleanup code directly